filteredQuery Failed to build search source

client.execute { search in "a/b" query filteredQuery filter ( termFilter("user.id", "some id")) }

play.api.Application$$anon$1: Execution exception[[SearchSourceBuilderException: Failed to build search source]]
    at play.api.Application$class.handleError(Application.scala:296) ~[play_2.11-2.3.1.jar:2.3.1]
    at play.api.DefaultApplication.handleError(Application.scala:402) [play_2.11-2.3.1.jar:2.3.1]
    at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$4.apply(PlayDefaultUpstreamHandler.scala:320) [play_2.11-2.3.1.jar:2.3.1]
    at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$4.apply(PlayDefaultUpstreamHandler.scala:320) [play_2.11-2.3.1.jar:2.3.1]
    at scala.Option.map(Option.scala:145) [scala-library-2.11.1.jar:na]
Caused by: org.elasticsearch.search.builder.SearchSourceBuilderException: Failed to build search source
    at org.elasticsearch.search.builder.SearchSourceBuilder.buildAsBytes(SearchSourceBuilder.java:721) ~[elasticsearch-1.2.1.jar:na]
    at org.elasticsearch.action.search.SearchRequest.source(SearchRequest.java:235) ~[elasticsearch-1.2.1.jar:na]
    at org.elasticsearch.action.search.SearchRequestBuilder.request(SearchRequestBuilder.java:1061) ~[elasticsearch-1.2.1.jar:na]
    at com.sksamuel.elastic4s.SearchDsl$SearchDefinition.build(SearchDsl.scala:74) ~[elastic4s_2.11-1.2.1.3.jar:1.2.1.3]
    at com.sksamuel.elastic4s.SearchDsl$SearchDefinition.build(SearchDsl.scala:67) ~[elastic4s_2.11-1.2.1.3.jar:1.2.1.3]
Caused by: java.lang.NullPointerException: null
    at org.elasticsearch.index.query.FilteredQueryBuilder.doXContent(FilteredQueryBuilder.java:72) ~[elasticsearch-1.2.1.jar:na]
    at org.elasticsearch.index.query.BaseQueryBuilder.toXContent(BaseQueryBuilder.java:66) ~[elasticsearch-1.2.1.jar:na]
    at org.elasticsearch.search.builder.SearchSourceBuilder.toXContent(SearchSourceBuilder.java:743) ~[elasticsearch-1.2.1.jar:na]
    at org.elasticsearch.search.builder.SearchSourceBuilder.buildAsBytes(SearchSourceBuilder.java:718) ~[elasticsearch-1.2.1.jar:na]
    at org.elasticsearch.action.search.SearchRequest.source(SearchRequest.java:235) ~[elasticsearch-1.2.1.jar:na]

--- a/src/main/scala/com/sksamuel/elastic4s/queries.scala
+++ b/src/main/scala/com/sksamuel/elastic4s/queries.scala
@@ -465,7 +465,7 @@ class DisMaxDefinition extends QueryDefinition {
 
 class FilteredQueryDefinition extends QueryDefinition {
   def builder = QueryBuilders.filteredQuery(_query, _filter).boost(_boost.toFloat)
-  private var _query: QueryBuilder = null
+  private var _query: QueryBuilder = QueryBuilders.matchAllQuery
   private var _filter: FilterBuilder = null
   private var _boost: Double = -1d
   def boost(boost: Double): FilteredQueryDefinition = {
@@ -473,7 +473,7 @@ class FilteredQueryDefinition extends QueryDefinition {
     this
   }
   def query(query: => QueryDefinition): FilteredQueryDefinition = {
-    _query = Option(query).map(_.builder).orNull
+    _query = Option(query).map(_.builder).getOrElse(_query)
     this
   }
   def filter(filter: => FilterDefinition): FilteredQueryDefinition = {

--- a/src/test/resources/json/search/search_default_query.json
+++ b/src/test/resources/json/search/search_default_query.json
@@ -1,0 +1,12 @@
+{
+  "filtered" : {
+    "query" : {
+      "match_all" : { }
+    },
+    "filter" : {
+      "term" : {
+        "singer" : "lemmy"
+      }
+    }
+  }
+}

--- a/src/test/scala/com/sksamuel/elastic4s/SearchDslTest.scala
+++ b/src/test/scala/com/sksamuel/elastic4s/SearchDslTest.scala
@@ -905,5 +905,11 @@ class SearchDslTest extends FlatSpec with MockitoSugar with JsonSugar with OneIn
     }
     req._builder.toString should matchJsonResource("/json/search/search_and_filter.json")
   }
+
+  it should "generate correct json for default filtered query" in {
+    val req = filteredQuery filter termFilter("singer", "lemmy")
+
+    req.builder.toString should matchJsonResource("/json/search/search_default_query.json")
+  }
 }
 

